monai/visualize/occlusion_sensitivity.py

```diff
@@ -142,139 +71,184 @@ class OcclusionSensitivity:
-        """
+            verbose: Use progress bar (if ``tqdm`` available).
+            mode: what should the occluded region be replaced with? If a float is given, that value will be used
+                throughout the occlusion. Else, ``gaussian``, ``mean_img`` and ``mean_patch`` can be supplied:
+
+                * ``gaussian``: occluded region is multiplied by 1 - gaussian kernel. In this fashion, the occlusion
+                  will be 0 at the center and will be unchanged towards the edges, varying smoothly between. When
+                  gaussian is used, a weighted average will be used to combine overlapping regions. This will be
+                  done using the gaussian (not 1-gaussian) as occluded regions count more.
+                * ``mean_patch``: occluded region will be replaced with the mean of occluded region.
+                * ``mean_img``: occluded region will be replaced with the mean of the whole image.
 
+            overlap: overlap between inferred regions. Should be in range 0<=x<1.
+            activate: if ``True``, do softmax activation if num_channels > 1 else do ``sigmoid``. If ``False``, don't do any
+                activation. If ``callable``, use callable on inferred outputs.
+
+        """
         self.nn_module = nn_module
-        self.upsampler = upsampler
-        self.pad_val = pad_val
         self.mask_size = mask_size
         self.n_batch = n_batch
-        self.stride = stride
-        self.per_channel = per_channel
         self.verbose = verbose
+        self.overlap = overlap
+        self.activate = activate
+        # mode
+        if isinstance(mode, str) and mode not in ("gaussian", "mean_patch", "mean_img"):
+            raise NotImplementedError
+        self.mode = mode
+
+    @staticmethod
+    def constant_occlusion(x: torch.Tensor, val: float, mask_size: tuple) -> Tuple[float, torch.Tensor]:
+        """Occlude with a constant occlusion. Multiplicative is zero, additive is constant value."""
+        ones = torch.ones((*x.shape[:2], *mask_size), device=x.device, dtype=x.dtype)
+        return 0, ones * val
+
+    @staticmethod
+    def gaussian_occlusion(x: torch.Tensor, mask_size) -> Tuple[torch.Tensor, float]:
+        """For Gaussian occlusion, Multiplicative is 1-Gaussian, additive is zero."""
+        kernel = torch.zeros((x.shape[1], *mask_size), device=x.device, dtype=x.dtype)
+        spatial_shape = kernel.shape[1:]
+        # all channels (as occluded shape already takes into account per_channel), center in spatial dimensions
+        center = [slice(None)] + [slice(s // 2, s // 2 + 1) for s in spatial_shape]
+        # place value of 1 at center
+        kernel[center] = 1.0
+        # Smooth with sigma equal to quarter of image, flip +ve/-ve so largest values are at edge
+        # and smallest at center. Scale to [0, 1].
+        gaussian = Compose(
+            [GaussianSmooth(sigma=[b // 4 for b in spatial_shape]), Lambda(lambda x: -x), ScaleIntensity()]
```

Comment:
why choose this sigma as `b//4`? this might need some comments

Reply:
Empirically I found this to give good results. Below I've attached imshows of the kernels for mask sizes of 15 and 31. I don't think this parameter should be offered to the user to change, the goal is simply to give a smoothly varying kernel. What do you think? I can add to the documentation that it is empirically determined and give the code to do the imshow?

![image](https://user-images.githubusercontent.com/33289025/193264770-a4380a3a-6ab9-4ebf-86f5-380c3ffcc69a.png)

Reply:
thanks, I think your comment here is enough as a kind of documentation... so marking this as resolved.

Reply:
Few more plots, sigmas of 1, 0.25, 0.1 and mask sizes of 15 and 31. Think you'll agree that 0.25 looks best.
![image](https://user-images.githubusercontent.com/33289025/193265838-df77d5d7-78a3-41f1-a831-2bf52b31e5c8.png)